crypto/src/mls/conversation/decrypt.rs

```diff
@@ -123,13 +124,19 @@ impl MlsConversation {
         callbacks: Option<&dyn CoreCryptoCallbacks>,
         restore_pending: bool,
     ) -> CryptoResult<MlsConversationDecryptMessage> {
-        // handles the crooked case where we receive our own commits.
-        // Since this would result in an error in openmls, we handle it here
-        if let Some(ct) = self.maybe_self_member_commit(&message)? {
-            return self.handle_self_member_commit(backend, ct).await;
-        }
-
-        let message = self.parse_message(backend, message).await?;
+        let message = match self.parse_message(backend, message.clone()).await {
+            // Handles the case where we receive our own commits.
+            Err(CryptoError::MlsError(MlsError::MlsMessageError(ProcessMessageError::InvalidCommit(
+                StageCommitError::OwnCommit,
+            )))) => {
+                if let Some(ct) = self.maybe_self_member_commit(&message)? {
+                    return self.handle_self_member_commit(backend, ct).await;
+                }
+                Err(CryptoError::InternalMlsError)
```

Comment:
This is not expected to be ever reached. Should we `panic!()` instead?

Reply:
If this is indeed unreachable, there's `unreachable!()`.